plutus-tx/src/PlutusTx/Enum.hs

```diff
@@ -0,0 +1,85 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# OPTIONS_GHC -fno-omit-interface-pragmas #-}
+module PlutusTx.Enum (Enum(..)) where
+
+import           PlutusTx.Builtins
+import           PlutusTx.String   ()
+import           Prelude           hiding (Enum (..), error)
+
+-- | Class 'Enum' defines operations on sequentially ordered types.
+class Enum a where
+  -- | the successor of a value.  For numeric types, 'succ' adds 1.
+  succ :: a -> a
+  -- | the predecessor of a value.  For numeric types, 'pred' subtracts 1.
+  pred :: a -> a
+  -- | Convert from an 'Integer'.
+  toEnum :: Integer -> a
+  -- | Convert to an 'Integer'.
+  fromEnum :: a -> Integer
+
+instance Enum Integer where
+  {-# INLINABLE succ #-}
+  succ x = addInteger x 1
+
+  {-# INLINABLE pred #-}
+  pred x = subtractInteger x 1
+
+  {-# INLINABLE toEnum #-}
+  toEnum x = x
+
+  {-# INLINABLE fromEnum #-}
+  fromEnum x = x
+
+instance Enum () where
+  {-# INLINABLE succ #-}
+  succ _ = error (trace "PlutusTx.Enum.().succ: bad argument")
```

Comment:
`traceError` exists although not a big deal

Reply:
Yeah, it's in `Prelude` module that imports `Enum`. Maybe it's better to move it to it's own module with other trace functions, smth like `Prelude.Trace`?